Bug in class DataClass when loading read-only files

## What is the problem?

- The load methods in the DataClass class fail when loading a read-only file.
```
../teaser/project.py:120: in __init__
    self.data = self.instantiate_data_class()
../teaser/project.py:136: in instantiate_data_class
    return DataClass()
../teaser/data/dataclass.py:56: in __init__
    self.load_tb_binding()
../teaser/data/dataclass.py:88: in load_tb_binding
    with open(self.path_tb, "r+") as f:
E   PermissionError: [Errno 13] Permission denied: 'TEASER/teaser/data/input/inputdata/TypeBuildingElements.json'
```

## Why do we want to solve it?

- For example, to be able to run teaser on clusters where the python libraries might be installed in a read only filesystem

## How do we want to solve it?

- The mode of `open` should be set to "r" instead of "r+"

--- a/teaser/data/dataclass.py
+++ b/teaser/data/dataclass.py
@@ -85,7 +85,7 @@ class DataClass(object):
         if self.path_tb.endswith("json"):
             if os.path.isfile(self.path_tb):
                 try:
-                    with open(self.path_tb, "r+") as f:
+                    with open(self.path_tb, "r") as f:
                         self.element_bind = json.load(
                             f, object_pairs_hook=collections.OrderedDict
                         )
@@ -101,7 +101,7 @@ class DataClass(object):
         if self.path_uc.endswith("json"):
             if os.path.isfile(self.path_uc):
                 try:
-                    with open(self.path_uc, "r+") as f:
+                    with open(self.path_uc, "r") as f:
                         self.conditions_bind = json.load(
                             f, object_pairs_hook=collections.OrderedDict
                         )
@@ -117,7 +117,7 @@ class DataClass(object):
         if self.path_mat.endswith("json"):
             if os.path.isfile(self.path_mat):
                 try:
-                    with open(self.path_mat, "r+") as f:
+                    with open(self.path_mat, "r") as f:
                         self.material_bind = json.load(
                             f, object_pairs_hook=collections.OrderedDict
                         )

